src/librustc_resolve/diagnostics.rs

```diff
@@ -948,6 +948,33 @@ use something_which_doesnt_exist;
 Please verify you didn't misspell the import's name.
 "##,
 
+E0434: r##"
+A dynamic environment was used inside a fn item. Example of erroneous code: 
+
+
+```compile_fail
+fn foo(){
+    let y = 5;
+    fn bar() -> u32 {
+        y // error: can't capture dynamic environment in a fn item; use the 
+          // || { ... } closure form instead.
+    }
+}
+```
+
+Functions do not capture local variables. 
+To fix this error, please replace the function with a closure. Example:
```

Comment:
You're limited to 80 columns, so returning to the next line isn't needed at the end of the previous sentence (it looks strange like this).